File: src/dataset/go/proverb_Exercism_go_ThomasZumsteg/gemini/GeminiFlash_proverb_v4.go
package proverb

import "fmt"
import "strings"

func Proverb(rhyme []string) []string {
	n := len(rhyme)
	if n == 0 {
		return []string{}
	}

	result := make([]string, 0, n)
	var sb strings.Builder
	for i := 0; i < n-1; i++ {
		sb.WriteString("For want of a ")
		sb.WriteString(rhyme[i])
		sb.WriteString(" the ")
		sb.WriteString(rhyme[i+1])
		sb.WriteString(" was lost.")
		result = append(result, sb.String())
		sb.Reset()
	}

	sb.WriteString("And all for the want of a ")
	sb.WriteString(rhyme[0])
	sb.WriteString(".")
	result = append(result, sb.String())

	return result
}

File: src/dataset/typescript/proverb_Exercism-typescript-shybyte/gemini/GeminiFlash_proverb_v4.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = proverb;
function proverb(...nouns) {
    const len = nouns.length;
    if (len === 0) {
        return "";
    }
    const lines = new Array(len);
    for (let i = 0; i < len - 1; i++) {
        lines[i] = `For want of a ${nouns[i]} the ${nouns[i + 1]} was lost.`;
    }
    lines[len - 1] = `And all for the want of a ${nouns[0]}.`;
    return lines.join('\n');
}
